homework-submission/react-hyf/react-hyf-client/src/components/HomeworkWeek3/HomeworkWeek3.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import Profile from './Profile';
+
+const url = 'https://uinames.com/api/?amount=10';
+
+class HomeworkWeek3 extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { data: [] };
+  }
+
+  componentDidMount() {
+    fetch(url)
+      .then(response => response.json())
+      .then(json => 
+        json.map((elem)=>
+          this.setState({
+            data: elem,
+          })
+        ) 
+    );
+  }
+
+  componentWillMount() {
+    document.addEventListener("keydown", this.onKeyPressed.bind(this));
+  }
+
+  onKeyPressed(e) {
+      if (e.keyCode === 32) {
+        this.componentDidMount();
```

Comment:
Nice! But you should extract the fetch to it's own function and call that function instead of calling `componentDidMount`